### PR TITLE
update aad changelog and readme

### PR DESCRIPTION
**Pull request by yiliuTo**
This PR updates changelog and readme for aad webapp features and some linguisitc modification.
Added feature introduction include: 
- deprecated configuration properties
- deprecate support of v1 endpoints.
- transitive membership

### CODE REVIEW

File: sdk/spring/azure-spring-boot-starter-active-directory/README.md [hunk 1]
```diff
@@ -27,16 +27,21 @@ With Spring Starter for Azure Active Directory, now you can get started quickly
...
-This package provides 2 ways to integrate with Spring Security and authenticate with Azure Active Directory.
+This package provides 2 ways to integrate with Spring Security and authenticate with Azure Active Directory, which are designed for scenarios of web application and resource server.
 
-The authorization flow is composed of 3 phrases:
+The authorization flow for web application includes:
+* Login with credentials by self-defined `azure` client registration and trigger **authorization code flow**. Application gets user's consent for all configured scopes except on-demand resources' scopes, and acquires an access token only for scopes of `openid, profile, offline_access, https://graph.microsoft.com/User.Read`.
+* When other resources are visited, associated clients will be loaded to trigger **refresh token flow** for authorization, and acquire an access token for configured scopes of that resource.
+* When on-demand resources are visited, associated clients will be loaded and trigger **authorization code flow** like the `azure` client.
+
+The authorization flow for resource server:
```
Comment (yiliuTo) [inline sdk/spring/azure-spring-boot-starter-active-directory/README.md:37]: Here and below introduction related with webapi will be udpated by @backwind1233

File: sdk/spring/azure-spring-boot-starter-active-directory/README.md [hunk 2]
```diff
@@ -27,16 +27,21 @@ With Spring Starter for Azure Active Directory, now you can get started quickly
...
 * Login with credentials and validate id_token from Azure AD 
 * Get On-Behalf-Of token and membership info from Azure AD Graph API
 * Evaluate the permission based on membership info to grant or deny access
 
 ### Group membership
-The way to get group relationship depends on the graph api used, the default to get membership is the direct group of the user. 
-To get all transitive relationships, the following configuration is required:
+The way this starter uses to load get users' membership depends on the configured membership URI. By default, the starter uses `https://graph.microsoft.com/v1.0/me/memberOf` to get direct membership of current user.
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter-active-directory/README.md:43]: `The way this starter uses to load users' membership`, Delete `get`

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot-starter-active-directory/CHANGELOG.md [hunk 3]
```diff
@@ -5,16 +5,39 @@
...
+- Support the use of `@RegisteredOAuth2AuthorizedClient` annotation to get `OAuth2AuthorizedClient`.
+- Support access control through users' membership information.
+- Support on-behalf-of flow in the resource server.
+- Provide AAD specific token validation of audience validation and issuer validation.
+- Expose a flag `isPersonalAccount` in `AzureOAuth2AuthenticatedPrincipal` to specify the account type in use: work account or personal account.
+- Enable loading transitive membership information from Microsoft Graph API.
+- Enable following `azure-spring-boot-starter-active-directory` configuration properties:
+    ```yaml
+    # Redirect URI of authorization server
+    azure.activedirectory.redirectUriTemplate
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter-active-directory/CHANGELOG.md:31]: `redirect-uri-template`. Same to other items.

File: sdk/spring/azure-spring-boot-starter-active-directory/README.md [hunk 4]
```diff
@@ -16,27 +16,38 @@ With Spring Starter for Azure Active Directory, now you can get started quickly
...
 ```xml
 <dependency>
     <groupId>com.azure.spring</groupId>
     <artifactId>azure-spring-boot-starter-active-directory</artifactId>
     <version>3.0.0-beta.1</version>
 </dependency>
+<dependency>
+  <groupId>org.springframework.security</groupId>
+  <artifactId>spring-security-oauth2-client</artifactId>
+</dependency>
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter-active-directory/README.md:31]: @backwind1233 , if it's used in resource-server, related dependency should be added.

File: sdk/spring/azure-spring-boot-starter/CHANGELOG.md [hunk 5]
```diff
@@ -1,7 +1,42 @@
...
+- Support the use of `@RegisteredOAuth2AuthorizedClient` annotation to get `OAuth2AuthorizedClient`.
+- Support access control through users' membership information.
+- Support on-behalf-of flow in the resource server.
+- Provide AAD specific token validation methods of audience validation and issuer validation.
+- Expose a flag `isPersonalAccount` in `AzureOAuth2AuthenticatedPrincipal` to specify the account type in use: work account or personal account.
+- Enable loading transitive membership information from Microsoft Graph API.
+- Enable following `azure-spring-boot-starter-active-directory` configuration properties:
+    ```yaml
+    # Redirect URI of authorization server
+    azure.activedirectory.redirectUriTemplate
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter/CHANGELOG.md:30]: Same here.

File: sdk/spring/azure-spring-boot/CHANGELOG.md [hunk 6]
```diff
@@ -3,17 +3,40 @@
...
+- Support the use of `@RegisteredOAuth2AuthorizedClient` annotation to get `OAuth2AuthorizedClient`.
+- Support access control through users' membership information.
+- Support on-behalf-of flow in the resource server.
+- Provide AAD specific token validation methods of audience validation and issuer validation.
+- Expose a flag `isPersonalAccount` in `AzureOAuth2AuthenticatedPrincipal` to specify the account type in use: work account or personal account.
+- Enable loading transitive membership information from Microsoft Graph API.
+- Enable following `azure-spring-boot-starter-active-directory` configuration properties:
+    ```yaml
+    # Redirect URI of authorization server
+    azure.activedirectory.redirectUriTemplate
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot/CHANGELOG.md:30]: Same here.

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot-starter-active-directory/CHANGELOG.md [hunk 7]
```diff
@@ -5,16 +5,39 @@
...
-- Provide some AAD specific token validation, such as audience validation, issuer validation.
-- Expose a flag in the `AzureOAuth2AuthenticatedPrincipal` to tell which account type is being used, work account or personal account.
-
+- Support consent of multiple client registrations during user login.
+- Support on-demand client registrations.
+- Support the use of `@RegisteredOAuth2AuthorizedClient` annotation to get `OAuth2AuthorizedClient`.
+- Support access control through users' membership information.
+- Support on-behalf-of flow in the resource server.
+- Provide AAD specific token validation of audience validation and issuer validation.
+- Expose a flag `isPersonalAccount` in `AzureOAuth2AuthenticatedPrincipal` to specify the account type in use: work account or personal account.
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-active-directory/CHANGELOG.md:26]: I believe `AzureOAuth2AuthenticatedPrincipal ` has been renamed.

File: sdk/spring/azure-spring-boot-starter/CHANGELOG.md [hunk 8]
```diff
@@ -1,7 +1,42 @@
...
+- Stop support of Azure Active Directory Endpoints.
 
+### New Features
+- Support consent of multiple client registrations during user login.
+- Support on-demand client registrations.
+- Support the use of `@RegisteredOAuth2AuthorizedClient` annotation to get `OAuth2AuthorizedClient`.
+- Support access control through users' membership information.
+- Support on-behalf-of flow in the resource server.
+- Provide AAD specific token validation methods of audience validation and issuer validation.
+- Expose a flag `isPersonalAccount` in `AzureOAuth2AuthenticatedPrincipal` to specify the account type in use: work account or personal account.
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter/CHANGELOG.md:25]: Same here.

File: sdk/spring/azure-spring-boot/CHANGELOG.md [hunk 9]
```diff
@@ -3,17 +3,40 @@
...
-- Provide some AAD specific token validation, such as audience validation, issuer validation.
-- Expose a flag in the `AzureOAuth2AuthenticatedPrincipal` to tell which account type is being used, work account or personal account.
-
+- Support consent of multiple client registrations during user login.
+- Support on-demand client registrations.
+- Support the use of `@RegisteredOAuth2AuthorizedClient` annotation to get `OAuth2AuthorizedClient`.
+- Support access control through users' membership information.
+- Support on-behalf-of flow in the resource server.
+- Provide AAD specific token validation methods of audience validation and issuer validation.
+- Expose a flag `isPersonalAccount` in `AzureOAuth2AuthenticatedPrincipal` to specify the account type in use: work account or personal account.
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot/CHANGELOG.md:25]: Same here.

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot-starter-active-directory/CHANGELOG.md (hunk 7 above)
Comment (yiliuTo) [reply sdk/spring/azure-spring-boot-starter-active-directory/CHANGELOG.md]: Thanks, I have updated them now

**Review by saragluna — APPROVED** [review]

**Review by chenrujun — APPROVED** [review]
LGTM.

**Review by chenrujun — APPROVED** [review]
LGTM.

File: sdk/spring/azure-spring-boot-starter-active-directory/README.md (hunk 1 above)
Comment (saragluna) [reply sdk/spring/azure-spring-boot-starter-active-directory/README.md:43]: @backwind1233 will we create a separate PR for this?

File: sdk/spring/azure-spring-boot-starter-active-directory/README.md [hunk 10]
```diff
@@ -16,34 +16,76 @@ With Spring Starter for Azure Active Directory, now you can get started quickly
...
-
-```yaml
-azure:
-  activedirectory:
-    graph-membership-uri: https://graph.microsoft.com/v1.0/me/transitiveMemberOf
-``` 
-
+### Configurable properties
+This starter provides following properties to be customized:
+* azure.activedirectory.client-id:
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-starter-active-directory/README.md:50]: Is it only me or does anyone else find this list of properties hard to read?

Comment (chenrujun) [reply sdk/spring/azure-spring-boot-starter-active-directory/README.md:50]: How about convert this to a table?

Comment (yiliuTo) [reply sdk/spring/azure-spring-boot-starter-active-directory/README.md:50]: what about we using properties format like:
```properties
# Registered application ID in Azure AD.
azure.activedirectory.client-id=
```

File: sdk/spring/azure-spring-boot-starter-active-directory/README.md (hunk 1 above)
Comment (backwind1233) [reply sdk/spring/azure-spring-boot-starter-active-directory/README.md:43]: @saragluna  I think so

File: sdk/spring/azure-spring-boot-starter-active-directory/README.md (hunk 10 above)
Comment (saragluna) [reply sdk/spring/azure-spring-boot-starter-active-directory/README.md:50]: I'll vote for the table.

**Pull request merged**